arviz/data/inference_data.py

```diff
@@ -324,6 +355,311 @@ def sel(
         else:
             return out
 
+    def isel(
+        self, groups=None, filter_groups=None, inplace=False, **kwargs,
+    ):
+        """Perform an xarray selection on all groups.
+
+        Loops groups to perform Dataset.isel(key=item)
+        for every kwarg if key is a dimension of the dataset.
+        One example could be performing a burn in cut on the InferenceData object
+        or discarding a chain. The selection is performed on all relevant groups (like
+        posterior, prior, sample stats) while non relevant groups like observed data are
+        omitted. See :meth:`xarray.Dataset.isel <xarray:xarray.Dataset.isel>`
```

Comment:
simplify to 

    :meth:`xarray:xarray.Dataset.isel`

the example is `idata.sel` is pedagogical to show how would one use custom hyperlink text.